src/main/java/Iris/ToDoList.java

```diff
@@ -0,0 +1,157 @@
+     */
+    public void remove(int index) {
+        if (index < 1 || index > list.size()) {
+            throw new IndexOutOfBoundsException("Task index is out of range.");
+        }
+        list.remove(index - 1);
+    }
+
+    /**
+     * Gets a task from the list by index.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the specified index.
+     * @throws IndexOutOfBoundsException If the index is out of range.
+     */
+    public Task get(int index) {
+        if (index < 1 || index > list.size()) {
+            throw new IndexOutOfBoundsException("Task index is out of range.");
+        }
+        return list.get(index - 1);
+    }
+
+    /**
+     * Marks a task as done by index.
+     *
+     * @param index The index of the task to be marked as done.
+     * @throws IndexOutOfBoundsException If the index is out of range.
+     */
+    public void mark(int index) {
+        if (index < 1 || index > list.size()) {
+            throw new IndexOutOfBoundsException("Task index is out of range.");
+        }
+        list.get(index - 1).markDone();
+    }
+
+    /**
+     * Marks a task as undone by index.
+     *
+     * @param index The index of the task to be marked as undone.
+     * @throws IndexOutOfBoundsException If the index is out of range.
+     */
+    public void unmark(int index) {
+        if (index < 1 || index > list.size()) {
+            throw new IndexOutOfBoundsException("Task index is out of range.");
+        }
+        list.get(index - 1).markUndone();
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return The number of tasks in the list.
+     */
+    public int size() {
+        return list.size();
+    }
+
+    /**
+     * Adds a task to the ToDoList based on the command and description provided.
+     *
+     * @param toDoList     The ToDoList to which the task will be added.
+     * @param command      The command indicating the type of task (e.g., "todo").
+     * @param description  The description of the task.
+     * @throws EmptyTaskDescriptorsException If the task description is empty.
+     */
+    public static void addTask(ToDoList toDoList, String command, String description)
+            throws EmptyTaskDescriptorsException {
+        if (description.isEmpty()) {
+            throw new EmptyTaskDescriptorsException();
+        }
+        Task task = null;
+        if (command.equalsIgnoreCase("todo")) {
+            task = new Todo(description);
+        } else if (command.equalsIgnoreCase("deadline")) {
+            String[] deadlineSections = description.split(" /by ", 2);
+            String name = deadlineSections[0];
+            String deadlineString = deadlineSections[1];
+            task = new Deadline(name, deadlineString);
+        } else if (command.equalsIgnoreCase("event")) {
+            String[] eventSections = description.split(" /from ", 2);
+            String name = eventSections[0];
```

Comment:
You can consider using .trim() when retrieving parts after splitting, helpful in avoiding any user-generated error caused by additional spaces.